core/src/test/java/org/apache/calcite/tools/PlannerTest.java

```diff
@@ -969,7 +969,7 @@ private void checkJoinNWay(int n) throws Exception {
     RelNode transform = planner.transform(0, traitSet, convert);
     assertThat(toString(transform),
         containsString(
-            "EnumerableMergeJoin(condition=[=($0, $5)], joinType=[inner])"));
+            "EnumerableHashJoin(condition=[=($0, $5)], joinType=[inner])"));
```

Comment:
Changes in this class rolls back some changes in https://github.com/apache/calcite/commit/bb484859e115596061c572b689a3b04349299458, because now we turns off top-down opt globally.

Reply:
TpcdsTest.testQuery17Plan also failed, which is expected. You can checkout TpcdsTest.java from any commit before I introduced top-down.

Reply:
Fixed.